massa-network-worker/src/node_worker.rs

```diff
@@ -207,74 +185,19 @@ impl NodeWorker {
-                                self.send_node_event(NodeEvent(self.node_id, NodeEventType::ReceivedPeerList(pl))).await;
-                            }
-                            Message::AskPeerList => {
-                                self.send_node_event(NodeEvent(self.node_id, NodeEventType::AskedPeerList)).await;
-                            }
-                            Message::Operations(operations) => {
-                                massa_trace!(
-                                    "node_worker.run_loop. receive Message::Operations: ",
-                                    {"node": self.node_id, "operations": operations}
-                                );
-                                //massa_trace!("node_worker.run_loop. receive Message::Operations", {"node": self.node_id, "operations": operations});
-                                self.send_node_event(NodeEvent(self.node_id, NodeEventType::ReceivedOperations(operations))).await;
-                            }
-                            Message::AskForOperations(operation_prefix_ids) => {
-                                massa_trace!(
-                                    "node_worker.run_loop. receive Message::AskForOperations: ",
-                                    {"node": self.node_id, "operation_ids": operation_prefix_ids}
-                                );
-                                //massa_trace!("node_worker.run_loop. receive Message::AskForOperations", {"node": self.node_id, "operations": operation_ids});
-                                self.send_node_event(NodeEvent(self.node_id, NodeEventType::ReceivedAskForOperations(operation_prefix_ids))).await;
-                            }
-                            Message::OperationsAnnouncement(operation_prefix_ids) => {
-                                massa_trace!("node_worker.run_loop. receive Message::OperationsBatch", {"node": self.node_id, "operation_prefix_ids": operation_prefix_ids});
-                                self.send_node_event(NodeEvent(self.node_id, NodeEventType::ReceivedOperationAnnouncements(operation_prefix_ids))).await;
-                            }
-                            Message::Endorsements(endorsements) => {
-                                massa_trace!("node_worker.run_loop. receive Message::Endorsement", {"node": self.node_id, "endorsements": endorsements});
-                                self.send_node_event(NodeEvent(self.node_id, NodeEventType::ReceivedEndorsements(endorsements))).await;
-                            }
-                            _ => {
-                                // TODO: Write a more user-friendly warning/logout after several consecutive fails? see #1082
-                                massa_trace!("node_worker.run_loop.self.socket_reader.next(). Unexpected message Warning", {});
-                            },
+                res = &mut node_reader_handle => {
```

Comment:
a select branch should also be added for node_writer_handle because if the writer exits but not the reader we won't capture it as it is

Reply:
my bad